File: app/src/main/java/com/relic/data/entities/CommentEntity.java
package com.relic.data.entities;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

@Entity
public class CommentEntity {
    public static final String MORE_CREATED = "more_author";

    @NonNull
    @PrimaryKey
    private String id;

    //public String replies;
    public String created;
    public String author;
    public String subreddit;

    // post parent id
    public String parentPostId;

    // parent comment id
    @ColumnInfo(name ="parentId")
    public String parent_id;

    @ColumnInfo(name = "body")
    public String body_html;

    public int score;

    public int platinum;
    public int gold;
    public int silver;

    @ColumnInfo(name = "authorFlairText")
    public String author_flair_text;
    @ColumnInfo(name = "authorFlairTextColor")
    public String author_flair_text_color;

    @ColumnInfo(name = "isSubmitter")
    public boolean is_submitter;
    @ColumnInfo(name = "scoreHidden")
    public boolean score_hidden;
    public int userUpvoted;
    public boolean saved;
    public boolean visited;

    public int replyCount;
    public int depth;
    public String replyLink;

    @ColumnInfo(name = "linkTitle")
    public String link_title;
    @ColumnInfo(name = "linkAuthor")
    public String link_author;

    public String editedDate;

    public CommentEntity(){}

    @NonNull
    public String getId() {
    return id;
    }

    public void setId(@NonNull String id) {
    this.id = id;
    }

    public float position;

    // for "more" values
    public String moreChildren;
}

File: app/src/main/java/com/relic/data/entities/CommentEntity.kt
package com.relic.data.entities

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

@Entity
class CommentEntity {

    @PrimaryKey
    var id: String = ""
    var fullName: String? = null
        get() = "t1_$id"

    //public String replies;
    var created: String? = null
    var author = ""
    var subreddit: String = ""

    // post parent id
    var parentPostId: String? = null

    // parent comment id
    @ColumnInfo(name = "parentFullname")
    var parent_id: String = ""

    @ColumnInfo(name = "body")
    var body_html: String? = null

    var score: Int = 0

    var platinum: Int = 0
    var gold: Int = 0
    var silver: Int = 0

    @ColumnInfo(name = "authorFlairText")
    var author_flair_text: String? = null
    @ColumnInfo(name = "authorFlairTextColor")
    var author_flair_text_color: String? = null

    @ColumnInfo(name = "isSubmitter")
    var submitter: Boolean = false
    @ColumnInfo(name = "scoreHidden")
    var score_hidden: Boolean = false
    var userUpvoted: Int = 0
    var saved: Boolean = false
    var visited: Boolean = false

    var replyCount: Int = 0
    var depth: Int = 0
    var replyLink: String? = null

    @ColumnInfo(name = "linkTitle")
    var link_title: String? = null
    @ColumnInfo(name = "linkAuthor")
    var link_author: String? = null

    var editedDate: String? = null

    var position: Float = 0.toFloat()

    // for "more" values
    var moreChildren: String? = null

    companion object {
        val MORE_CREATED = "more_author"
    }
}
